client/src/Stores/MicrosoftLearn.store.ts

```diff
@@ -155,4 +233,130 @@ export class MicrosoftLearnStore extends ChildStore {
   private scoreRegex(testPhrase: string, exp: RegExp, score = 1): number {
     return exp.test(testPhrase) ? score : 0;
   }
-}
+
+    // O(|filtertype|)
+  private removeExtrasFromSelected = (type: FilterType) => {
+    if(type===FilterType.Product){
+    
+          let parentProducts = Array.from(this.productMap.keys());
+  
+          // CASE 1: check for parent products which are there in selectedFilter but not in displayFilter. Remove all of them from selectedFilter, along with all their children.
+          // CASE 2: check for individual product filters whose parent is not in selected filter, but it is, and the current state of displayFilter does not. Remove them from selectedFilter.
+  
+          let prevSelected = this.selectedFilters.get(type);
+          let newDisplay = this.displayFilters.get(type);
+          let removeProducts: string[] = []
+          parentProducts?.forEach(item=>{
+          if(newDisplay && prevSelected){
+            if(prevSelected.includes(item.id) && !newDisplay.includes(item.id))
+                  removeProducts.push(item.id);
+          }
+          })
+          let productIter = this.catalog?.products.values();
+          let _n = 0;
+          while(this.catalog?.products.size && _n<this.catalog?.products.size){
+            let p = productIter?.next().value;
+            if((removeProducts.includes(p.parentId) ||
+            (!prevSelected?.includes(p.parentId) && prevSelected?.includes(p.id) && !newDisplay?.includes(p.id)))){
+                removeProducts.push(p.id);
+            }
+            _n=_n+1;
+          }
+          let newSelected = prevSelected?.filter(item => !removeProducts.includes(item) );
+        if(newSelected){

```

Comment:
No, I didn't mean otherwise. The question was more on the lines of "When do you feel the check would evaluate to `false`"? Is it when `newSelected` is null, undefined, or empty array? Or anything else that I am missing?

Reply:
Oh, I see. I wanted to check for `undefined` entries, but the above condition would hold for empty values as well, which I did not want. I have updated the same now.

Reply:
I believe it's the contrary, the above condition won't hold for empty arrays. I believe it'll have a faster resolution if we discuss this offline.